core/src/org/javarosa/core/model/condition/EvaluationContext.java

```diff
@@ -259,141 +261,141 @@ private void expandReferenceAccumulator(TreeReference sourceRef, DataInstance so
-                        Object o;
-                        o = xpe.eval(sourceInstance, evalContext);
-
-                        //There's a special case here that can't be handled by syntactic sugar.
-                        //If the result of a predicate expression is an Integer, we need to
-                        //evaluate whether that value is equal to the current position context
-
-                        o = XPathFuncExpr.unpack(o);
-
-
-
-                        if (o instanceof Double) {
-                            //The spec just says "number" for when to use
-                            //this, so I think this is ok? It's not clear
-                            //what to do with a non-integer. It's possible
-                            //we are not supposed to round.
-                            int intVal = XPathFuncExpr.toInt(o).intValue();
-                            passed = (intVal == positionContext[predIndex]);
-                        } else if (o instanceof Boolean) {
-                            passed = ((Boolean)o).booleanValue();
-                        } else {
-                            //???
-                        }
-
-                        if (!passed) {
-                            passedAll = false;
-                            break;
-                        }
-                    }
-                    if (predicateEvaluationProgress != null) {
-                        predicateEvaluationProgress[0]++;
+    /**
+     * Load viable references that are children of the element argument.
+     *
+     * @param node element of which to collect child references
+     * @param name
+     * @param mult
+     * @param includeTemplates
+     * @return
+     */
+    private Vector<TreeReference> loadReferencesChildren(AbstractTreeElement node,
+                                                         String name,
+                                                         int mult,
+                                                         boolean includeTemplates) {
+        Vector<TreeReference> childSet = new Vector<TreeReference>();
+        if (node.hasChildren()) {
+            if (mult == TreeReference.INDEX_UNBOUND) {
+                int count = node.getChildMultiplicity(name);
+                for (int i = 0; i < count; i++) {
+                    AbstractTreeElement child = node.getChild(name, i);
+                    if (child != null) {
+                        childSet.addElement(child.getRef());
+                    } else {
+                        throw new IllegalStateException("Missing or non-sequential nodes expanding a reference");
                     }
-                    if (passedAll) {
-                        expandReferenceAccumulator(sourceRef, sourceInstance, treeRef, refs, includeTemplates);
+                }
+                // TODO: this could be lifted out a conditional level, right? -- PLM
```

Comment:
Please double check, before I lift the "if (includeTemplates)" body out of the "if (mult == TreeReference.INDEX_UNBOUND)" body